ntbs-service/Services/LegacySearchService.cs

```diff
@@ -0,0 +1,118 @@
+using System;
+using System.Collections.Generic;
+using System.Data.SqlClient;
+using System.Linq;
+using System.Threading.Tasks;
+using ntbs_service.Models;
+using Dapper;
+using Microsoft.Extensions.Configuration;
+using ntbs_service.Models.Enums;
+using ntbs_service.Helpers;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Authorization;
+using System.Security.Claims;
+
+namespace ntbs_service.Services
+{
+    
+    public interface ILegacySearchService
+    {
+        Task<(IEnumerable<Notification> notifications, int count)> Search(int offset, int pageSize);
+    }
+
+    public class LegacySearchService : ILegacySearchService
+    {
+        const string Query = @"
+            SELECT *
+            FROM Notifications n 
+            LEFT JOIN Addresses addrs ON addrs.OldNotificationId = n.OldNotificationId
+            LEFT JOIN Demographics dmg ON dmg.OldNotificationId = n.OldNotificationId
+            ORDER BY n.NotificationDate DESC, n.OldNotificationId
+            OFFSET @Offset ROWS
+            FETCH NEXT @Fetch ROWS ONLY";
+
+        const string CountQuery = @"
+            SELECT COUNT(*)
+            FROM Notifications n";
+
+        private readonly IAuthorizationService _authorizationService;    
+
+        private readonly string connectionString;
+
+        public LegacySearchService(IConfiguration _configuration, IAuthorizationService authorizationService)
+        {
+            connectionString = _configuration.GetConnectionString("migration");
+            _authorizationService = authorizationService;
+        }
+
+        public async Task<(IEnumerable<Notification> notifications, int count)> Search(int offset, int numberToFetch)
+        {
+            IEnumerable<dynamic> results;
+            int count;
+            using (var connection = new SqlConnection(connectionString))
+            {
+                connection.Open();
+
+                results = await connection.QueryAsync(Query, new
+                {
+                    Offset = offset,
+                    Fetch = numberToFetch
+                });
+
+                count = (await connection.QueryAsync<int>(CountQuery)).Single();
+            }
+            return (results.Select(AsNotification), count);
+        }
+
+        private static Notification AsNotification(dynamic result)

```

Comment:
I'm weary of it returning an actual notification - it comes packed with a lot of overhead that is simply not needed just for search results. Perhaps we need a `SearchResult` interface?

Reply:
The banner model might be a good starting point for it.. 🤔